kadasrouting/core/canvaslayersaver.py

```diff
@@ -0,0 +1,88 @@
+from qgis.core import (
+    QgsVectorLayer,
+    QgsCoordinateReferenceSystem,
+    QgsWkbTypes,
+    QgsProject,
+    QgsFeature,
+    QgsSingleSymbolRenderer,
+    QgsGeometry,
+)
+from PyQt5.QtGui import QColor
+from kadasrouting.utilities import transformToWGS
+
+
+class CanvasLayerSaver():
+    def __init__(
+                self,
+                name,
+                features=None,
+                crs=QgsCoordinateReferenceSystem(4326),
+                color=QColor('lightGray'),
+                style=None
+            ):
+        self.name = name
+        self.color = color
+        self.transformer = transformToWGS(crs)
+        self.features = features
+        self.style = style
+        self.addPolygonLayer()
+
+    def addPolygonLayer(self):
+        try:
+            # FIXME: we do not consider if there are several layers with the same name here
+            existinglayer = QgsProject.instance().mapLayersByName(self.name)[0]
+            QgsProject.instance().removeMapLayer(existinglayer.id())
+        except IndexError:
+            # if such layer does not exist we can pass over
+            pass
+        # arbitrary decision taken: WGS84 for these layers
+        self.layer = QgsVectorLayer(
+            "Polygon?crs=epsg:4326&field=centerx:double&field=centery:double&field=interval:double",
+            self.name,
+            "memory",
+        )
+        pr = self.layer.dataProvider()
+        if self.features:
+            for feature in self.features:
+                print(feature)
+                if isinstance(feature, QgsFeature):
+                    geom = feature.geometry()
+                    feature.setGeometry(self.reprojectToWGS84(geom))
+                    pr.addFeatures([feature])
+                elif isinstance(feature, QgsGeometry):
+                    qgsFeature = QgsFeature()
```

Comment:
I think this should be `qgsfeature = QgsFeature(layer)`, so the feature that is added has the same attributes as the layer. Otherwise, I am not sure if the feature will be added. It will not raise an exception, but the `addFeatures` will return false to indicate failure to add the feature.

Reply:
Actually I just tested and there is no implementation that take a layer as parameter of the constructor apparently. Or at least not with QgsVectorLayer